analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Collections.cs

```diff
@@ -49,7 +47,7 @@ private ProgramState LearnBranchingCollectionConstraint(ProgramState state, IBin
 
         ISymbol InstanceOfCountProperty(IOperation operation) =>
             operation.AsPropertyReference() is { Instance: { } instance, Property.Name: nameof(Array.Length) or nameof(List<int>.Count) }
-            && instance.Type.DerivesOrImplementsAny(EmptyCollectionsShouldNotBeEnumeratedBase.TrackedCollectionTypes)
+            && instance.Type.DerivesOrImplementsAny(CollectionTracker.CollectionTypes)
```

Comment:
Consider moving all code from this class to `CollectionTracker` eventually (probably not in this PR).

Reply:
Yeah, will do!